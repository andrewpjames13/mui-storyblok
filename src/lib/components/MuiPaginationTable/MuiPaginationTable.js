import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Table } from '@material-ui/core';
import StoryBlok from '../../utils/Storyblok';

import MuiPaginationTableHead from './components/MuiPaginationTableHead/MuiPaginationTableHead';
import MuiPaginationTableBody from './components/MuiPaginationTableBody/MuiPaginationTableBody';
import MuiPaginationTableFooter from './components/MuiPaginationTableFooter/MuiPaginationTableFooter';

const MuiPaginationTable = ({
  rootClass,
  padding,
  size,
  stickyHeader,
  tHead,
  tBody,
  labelRowsPerPage,
  rowsPerPageOptions,
}) => {
  const styles = StoryBlok.arrayToMuiStyles(rootClass);
  const [numRows, setNumRows] = useState(0);
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(parseInt(rowsPerPageOptions[0], 10));

  const handleRowNums = (totalRows) => {
    setNumRows(totalRows);
  };

  const handlePageChange = (e, newPage) => {
    setPage(newPage);
  };

  const handleChangeRows = (e) => {
    const { value } = e.target;
    setRowsPerPage(parseInt(value, 10));
    setPage(0);
  };

  return (
    <Table
      className={styles.root}
      padding={padding}
      size={size}
      stickyHeader={stickyHeader}
    >
      <MuiPaginationTableHead {...tHead[0]} />
      <MuiPaginationTableBody
        {...tBody[0]}
        handleRowNums={handleRowNums}
        numRows={numRows}
        rowsPerPage={parseInt(rowsPerPage, 10)}
        page={page}
      />
      <MuiPaginationTableFooter
        labelRowsPerPage={labelRowsPerPage}
        rowsPerPageOptions={rowsPerPageOptions}
        numRows={numRows}
        page={page}
        handleChangeRows={handleChangeRows}
        rowsPerPage={parseInt(rowsPerPage, 10)}
        handlePageChange={handlePageChange}
      />
    </Table>
  );
};

export default MuiPaginationTable;

MuiPaginationTable.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: 'default' | 'checkbox' | 'none'
   * Allows TableCells to inherit padding of the Table.
   */
  padding: PropTypes.string,
  /**
   * mui props: 'small' | 'medium'
   * Allows TableCells to inherit size of the Table.
   */
  size: PropTypes.string,
  /**
   * mui props: true | false
   * set the header sticky. (It doesn't work with IE 11)
   */
  stickyHeader: PropTypes.bool,

  /**  MuiPaginationTableHead limit of 1 */
  tHead: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
  /**  MuiPaginationTableBody limit of 1 */
  tBody: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
  /**
   * mui prop: 'Rows per page:'
   * Customize the rows per page label.
   */
  labelRowsPerPage: PropTypes.node,
  /**
   * mui prop: [10, 25, 50, 100]
   * Customizes the options of the rows per page select field.
   * If less than two options are available, no select field will be displayed.
   */
  rowsPerPageOptions: PropTypes.arrayOf(PropTypes.number),
};

MuiPaginationTable.defaultProps = {
  padding: 'default',
  size: 'medium',
  stickyHeader: false,
  rootClass: [],
  labelRowsPerPage: 'Rows per page:',
  rowsPerPageOptions: [10, 25, 50, 100],
};
