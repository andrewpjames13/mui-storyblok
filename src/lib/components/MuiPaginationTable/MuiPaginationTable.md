MuiPaginationTable medium example (default):

```js
const tHead = [{
    component: 'MuiPaginationTableHead',
    content: [{
      component: 'MuiPaginationTableRow',
      content: [{
        component: 'MuiTableCell',
        size: 'large',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            content: 'Table Header 1',
          }],
        }],
      },
      {
        component: 'MuiTableCell',
        size: 'large',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            content: 'Table Header 2',
          }],
        }],
      },
      {
        component: 'MuiTableCell',
        size: 'large',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            content: 'Table Header 3',
          }],
        }],
      }],
    }],
  }];

  const tRow = {
    component: 'MuiPaginationTableRow',
    content: [
      {
        component: 'MuiTableCell',
        size: 'large',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                content: 'Table Body 1',
              },
            ],
          },
        ],
      },
      {
        component: 'MuiTableCell',
        size: 'large',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                content: 'Table Body 2',
              },
            ],
          },
        ],
      },
      {
        component: 'MuiTableCell',
        size: 'large',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                content: 'Table Body 3',
              },
            ],
          },
        ],
      },
    ],
  };

  const tableBody = [];

  for (let i = 0; i < 20; i++) {
   tableBody.push(tRow);
  };

  const tBody = [
    {
      component: 'MuiPaginationTableBody',
      content: [...tableBody],
    },
  ];

  <MuiPaginationTable
    tHead={tHead}
    tBody={tBody}
    size='medium'
    rowsPerPageOptions={[10, 25, 50, 100]}
  />
```

MuiPaginationTable small example:

```js
const tHead = [{
    component: 'MuiPaginationTableHead',
    content: [{
      component: 'MuiPaginationTableRow',
      content: [{
        component: 'MuiTableCell',
        size: 'small',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            content: 'Table Header 1',
          }],
        }],
      },
      {
        component: 'MuiTableCell',
        size: 'small',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            content: 'Table Header 2',
          }],
        }],
      },
      {
        component: 'MuiTableCell',
        size: 'small',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            content: 'Table Header 3',
          }],
        }],
      }],
    }],
  }];

  const tRow = {
    component: 'MuiPaginationTableRow',
    content: [
      {
        component: 'MuiTableCell',
        size: 'small',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                content: 'Table Body 1',
              },
            ],
          },
        ],
      },
      {
        component: 'MuiTableCell',
        size: 'small',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                content: 'Table Body 2',
              },
            ],
          },
        ],
      },
      {
        component: 'MuiTableCell',
        size: 'small',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                content: 'Table Body 3',
              },
            ],
          },
        ],
      },
    ],
  };

  const tableBody = [];

  for (let i = 0; i < 20; i++) {
   tableBody.push(tRow);
  };

  const tBody = [
    {
      component: 'MuiPaginationTableBody',
      content: [...tableBody],
    },
  ];

  <MuiPaginationTable
    tHead={tHead}
    tBody={tBody}
    size='small'
    rowsPerPageOptions={[5, 10, 25, 50]}
  />
```