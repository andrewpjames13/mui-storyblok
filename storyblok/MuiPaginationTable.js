module.exports = {
  "name": "MuiPaginationTable",
  "display_name": null,
  "updated_at": "2020-05-06T14:54:12.501Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "description": "* stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component."
    },
    "padding": {
      "type": "text",
      "description": "* mui props: 'default' | 'checkbox' | 'none'\n* Allows TableCells to inherit padding of the Table.",
      "default_value": "default"
    },
    "size": {
      "type": "text",
      "description": "* mui props: 'small' | 'medium'\n* Allows TableCells to inherit size of the Table.",
      "default_value": "medium"
    },
    "stickyHeader": {
      "type": "boolean",
      "description": "* mui props: true | false\n* set the header sticky. (It doesn't work with IE 11)"
    },
    "tHead": {
      "type": "bloks",
      "restrict_components": true,
      "required": true,
      "component_whitelist": [
        "MuiPaginationTableHead"
      ]
    },
    "tBody": {
      "type": "bloks",
      "required": true,
      "maximum": "",
      "restrict_components": true,
      "component_whitelist": [
        "MuiPaginationTableBody"
      ]
    },
    "labelRowsPerPage": {
      "type": "text",
      "max_length": "",
      "description": "* mui prop: 'Rows per page:'\n* Customize the rows per page label.",
      "default_value": "Rows per page:"
    },
    "rowsPerPageOptions": {
      "type": "options",
      "options": [
        {
          "value": "10",
          "name": "rows10"
        },
        {
          "value": "25",
          "name": "rows25"
        },
        {
          "value": "50",
          "name": "rows50"
        },
        {
          "value": "100",
          "name": "rows100"
        }
      ],
      "description": "   * Customizes the options of the rows per page select field.\n   * If less than two options are available, no select field will be displayed.",
      "default_value": ""
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiPaginationTable",
  "component_group_uuid": null
}