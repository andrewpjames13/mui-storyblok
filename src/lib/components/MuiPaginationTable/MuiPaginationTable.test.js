import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiPaginationTable from './MuiPaginationTable';

const row = {
  component: 'MuiPaginationTableRow',
  content: [
    {
      component: 'MuiTableCell',
      content: [
        {
          component: 'MuiTypography',
          content: [
            {
              component: 'MuiText',
              text: 'wooo',
            },
          ],
        },
      ],
    },
  ],
};

const rows = [];
do {
  // eslint-disable-next-line no-plusplus
  rows.push(row);
} while (rows.length < 12);

function setup(tBody = rows) {
  const props = {
    tHead: [
      {
        content: [
          {
            component: 'MuiPaginationTableRow',
            content: [
              {
                component: 'MuiTableCell',
                content: [
                  {
                    component: 'MuiTypography',
                    content: [
                      {
                        component: 'MuiText',
                        text: 'wooo',
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
    tBody: [
      {
        component: 'MuiPaginationTableBody',
        content: tBody,
      },
    ],
  };
  const comp = mount(<MuiPaginationTable {...props} />);
  return { comp, props };
}


describe('<MuiPaginationTable />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiPaginationTable {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiPaginationTable', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('should display the correct amount of rows after next/previous button clicked', () => {
    const { comp } = setup();
    const btn = comp.find('[data-testid="nextButton"]');
    const previousBtn = comp.find('[data-testid="previousButton"]');
    btn.first().simulate('click');
    const doesIncludeNext = comp.text().includes('11-12 of 12');
    expect(doesIncludeNext).toEqual(true);
    previousBtn.first().simulate('click');
    const doesIncludePrevious = comp.text().includes('1-10 of 12');
    expect(doesIncludePrevious).toEqual(true);
  });

  it('should display the correct amount of rows after first/last button clicked', () => {
    const newRows = [];
    do {
      newRows.push(row);
    } while (newRows.length < 50);

    const { comp } = setup(newRows);
    const lastBtn = comp.find('[data-testid="lastButton"]');
    const firstBtn = comp.find('[data-testid="firstButton"]');
    lastBtn.first().simulate('click');
    const doesIncludeLast = comp.text().includes('41-50 of 50');
    expect(doesIncludeLast).toEqual(true);
    firstBtn.first().simulate('click');
    const doesIncludeFirst = comp.text().includes('1-10 of 50');
    expect(doesIncludeFirst).toEqual(true);
  });
});
