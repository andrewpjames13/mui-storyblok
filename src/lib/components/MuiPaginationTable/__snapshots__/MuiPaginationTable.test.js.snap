// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiPaginationTable /> snapshot 1`] = `
<table
  className="MuiTable-root makeStyles-root-230"
  role={null}
>
  <thead
    className="MuiTableHead-root makeStyles-root-231"
    role={null}
  >
    <tr
      className="MuiTableRow-root makeStyles-root-232 MuiTableRow-head"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head makeStyles-root-233"
        scope="col"
      >
        <div
          className="MuiTypography-root makeStyles-root-234 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-235 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root makeStyles-root-236"
    role={null}
  >
    <tr
      className="MuiTableRow-root makeStyles-root-237"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-238"
      >
        <div
          className="MuiTypography-root makeStyles-root-239 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-240 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-241"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-242"
      >
        <div
          className="MuiTypography-root makeStyles-root-243 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-244 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-245"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-246"
      >
        <div
          className="MuiTypography-root makeStyles-root-247 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-248 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-249"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-250"
      >
        <div
          className="MuiTypography-root makeStyles-root-251 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-252 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-253"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-254"
      >
        <div
          className="MuiTypography-root makeStyles-root-255 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-256 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-257"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-258"
      >
        <div
          className="MuiTypography-root makeStyles-root-259 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-260 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-261"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-262"
      >
        <div
          className="MuiTypography-root makeStyles-root-263 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-264 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-265"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-266"
      >
        <div
          className="MuiTypography-root makeStyles-root-267 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-268 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-269"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-270"
      >
        <div
          className="MuiTypography-root makeStyles-root-271 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-272 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root makeStyles-root-273"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-274"
      >
        <div
          className="MuiTypography-root makeStyles-root-275 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-276 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </td>
    </tr>
  </tbody>
  <tfoot
    className="MuiTableFooter-root makeStyles-root-277"
    role={null}
  >
    <tr
      className="MuiTableRow-root MuiTableRow-footer"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root makeStyles-root-278"
        colSpan={1000}
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            className="MuiTablePagination-spacer"
          />
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            Rows per page:
          </p>
          <div
            className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-label="Rows per page:"
              className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              10
            </div>
            <input
              onAnimationStart={[Function]}
              type="hidden"
              value={10}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            0-0 of 0
          </p>
          <div
            className="makeStyles-root-279"
          >
            <button
              aria-label="first page"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              data-testid="firstButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                  />
                  <path
                    d="M24 24H0V0h24v24z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="previous page"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              data-testid="previousButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="next page"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              data-testid="nextButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="last page"
              className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
              data-testid="lastButton"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tfoot>
</table>
`;
